### PR TITLE
Update the main README, remove native/README

### LINKED ISSUE #206: Revise & update READMEs' content
We have multiple READMEs in the codebase. One in the root, another in `native` and one in `test`. All have some requirements listed and some steps relevant for the corresponding directory.

- [x] revise and update requirements
    * removed dependencies of dependencies (e.g., m4 is needed by httpd)
- [x] minimize duplication
    * removed `native/README.md`, it duplicated docs.modcluster.io
- [x] update (or remove) steps we have in the documentation already
    * see the above + added links to docs

### PR DESCRIPTION
**Pull request by jajik**
close #206 

Summary: 
* dependencies of dependencies were dropped from the tools list
* native/README.md was removed – we have it in our docs properly updated
* link added for the doxygen docs online

### CODE REVIEW

**Comment by jajik** [comment]
I left `test/README.md` as is, we do not have a section for it on the website and it looks up-to-date.

File: README.md [hunk 1]
```diff
@@ -72,11 +77,17 @@ You can also get additional information in case you compile mod_proxy_cluster wi
...
 search, but bear in mind that this may affect the performance.
 
-Doxygen documentation
+Documentation
 ---------------------
 
+The documentation can be found on the project's site – [docs.modcluster.io](https://docs.modcluster.io) – where
+you can find Getting started guide and documentation for the worker side (Java) part – mod_cluster. 
+
+### Doxygen api docs
```
Comment (rhusar) [inline README.md:86]: s/api/API

**Review by rhusar — CHANGES_REQUESTED** [review]
Looks good, just a nitpick.

File: README.md (hunk 1 above)
Comment (jajik) [reply README.md]: Fixed, thanks.

**Review by rhusar — APPROVED** [review]
LGTM now.

**Comment by jajik** [comment]
(rebase)

**Comment by rhusar** [comment]
Not really needed as this doesn't affect the tests (but we don't do conditional CI).

**Comment by rhusar** [comment]
Thanks @jajik

**Comment by jajik** [comment]
> Not really needed as this doesn't affect the tests (but we don't do conditional CI).

It depends on who's merging :smile: 

Thanks for the review & merge.

**Comment by rhusar** [comment]
> > Not really needed as this doesn't affect the tests (but we don't do conditional CI).
> 
> It depends on who's merging 😄

haha, true, maybe we should reconsider that requirement to be up to date..

**Pull request merged**